projects/plugins/debug-helper/modules/class-idc-simulator.php

```diff
@@ -0,0 +1,238 @@
+			'jetpack-debug-tools',
+			'IDC Simulator',
+			'IDC Simulator',
+			'manage_options',
+			'idc-simulator',
+			array( $this, 'render_ui' ),
+			99
+		);
+	}
+
+	/**
+	 * Render UI.
+	 */
+	public function render_ui() {
+		$settings = $this->get_stored_settings();
+		?>
+		<h1>IDC Simulator 😱!</h1>
+		<p>Cause an IDC on your site without having to clone it</p>
+
+		<h2>Current ICD simulation settings:</h2>
+		<p>Site URL spoofing: <?php echo esc_html( $settings['idc_simulation'] ); ?></p>
+		<p>Site URL value: <?php echo esc_html( $settings['idc_siteurl'] ); ?></p>
+
+		<form action="<?php echo esc_url( admin_url( 'admin-post.php' ) ); ?>" method="post">
+
+			<table class="form-table" role="presentation">
+			<tbody>
+			<tr>
+				<th scope="row">IDC Simulation</th>
+				<td>
+					<fieldset><legend class="screen-reader-text"><span>Site URL Spoofing</span></legend>
+					<label><input type="radio" name="idc_simulation" value="1" <?php echo ( $settings['idc_simulation'] ? 'checked="checked"' : '' ); ?>> enabled</label><br>
+					<label><input type="radio" name="idc_simulation" value="0" <?php echo ( ! $settings['idc_simulation'] ? 'checked="checked"' : '' ); ?>> disabled</label><br>
+					</fieldset>
+				</td>
+			</tr>
+			<tr>
+				<th scope="row"><label for="idc_siteurl">Spoof URL</label></th>
+				<td><input name="idc_siteurl" type="text" id="blogname" value="<?php echo esc_attr( $settings['idc_siteurl'] ); ?>" class="regular-text"></td>
+			</tr>
+
+			</tbody>
+			</table>
+
+			<input type="hidden" name="action" value="store_current_options">
+			<?php wp_nonce_field( 'store-current-options' ); ?>
+			<input type="submit" value="Store these options" class="button button-primary">
+		</form>
+
+		<hr>
+		<?php
+	}
+
+	/**
+	 * Store options.
+	 */
+	public function admin_post_store_current_options() {
+		check_admin_referer( 'store-current-options' );
+		$this->notice_type = 'store-options';
+
+		update_option(
+			self::STORED_OPTIONS_KEY,
+			array(
+				'idc_siteurl'    => $_POST['idc_siteurl'],
+				'idc_simulation' => $_POST['idc_simulation'],
+			)
+		);
+
+		$this->admin_post_redirect_referrer();
+	}
+
+	/**
+	 * Retrieves the stored IDC options.
+	 *
+	 * @return array
+	 */
+	public function get_stored_settings() {
+		return get_option( self::STORED_OPTIONS_KEY );
+	}
+
+	/**
+	 * Clears all stored option values.
+	 */
+	public function clear_stored_options() {
+		delete_option( self::STORED_OPTIONS_KEY );
+	}
+
+	/**
+	 * Just redirects back to the referrer. Keeping it DRY.
+	 */
+	public function admin_post_redirect_referrer() {
+		if ( wp_get_referer() ) {
+			wp_safe_redirect(
+				add_query_arg(
+					array(
+						'notice' => $this->notice_type,
+						'nonce'  => wp_create_nonce( 'jetpack_debug_broken_token_admin_notice' ),
+					),
+					wp_get_referer()
+				)
+			);
+		} else {
+			wp_safe_redirect( get_home_url() );
+		}
+	}
+
+	/**
+	 * Displays an admin notice...
+	 */
+	public function render_admin_notice() {
```

Comment:
Is `render_admin_notice()` used? I see a different notice ("_Success! Backup of the connection options stored safely._") when I save the IDC simulation settings.

Reply:
You're right, thank you! This was copypasta from the other module - I have removed it in favor of a simplified notice.